### PR TITLE
Finish rule sssd_offline_cred_expiration

### PR DESCRIPTION
**Pull request by matusmarhefka**
#### Description:

Added a bash remediation and test coverage for the rule sssd_offline_cred_expiration.

#### Rationale:

This rule is a part of Fedora OSPP profile.

### CODE REVIEW

File: linux_os/guide/services/sssd/sssd_offline_cred_expiration/oval/shared.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
   </ind:textfilecontent54_test>
   <ind:textfilecontent54_object id="obj_sssd_offline_cred_expiration" version="1">
     <ind:filepath>/etc/sssd/sssd.conf</ind:filepath>
-    <ind:pattern operation="pattern match">^\[pam]([^\n]*\n+)+?offline_credentials_expiration[\s]+=[\s]+1$</ind:pattern>
+    <ind:pattern operation="pattern match">^\[pam](?:[^\n]*\n+)+?offline_credentials_expiration[\s]+=[\s]+1$</ind:pattern>
```
Comment (redhatrises) [inline linux_os/guide/services/sssd/sssd_offline_cred_expiration/oval/shared.xml]: @matusmarhefka why this change?

Comment (matusmarhefka) [reply linux_os/guide/services/sssd/sssd_offline_cred_expiration/oval/shared.xml]: @redhatrises So why would you want to capture this group? It just doesn't make sense to me and that is why I rather set it as a group without capture.

Comment (jan-cerny) [reply linux_os/guide/services/sssd/sssd_offline_cred_expiration/oval/shared.xml]: It's not needed to do that, the rule works without that, but it's better.

File: tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/comment.fail.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+# profiles = xccdf_org.ssgproject.content_profile_ospp
+
+SSSD_CONF="/etc/sssd/sssd.conf"
```
Comment (dahaic) [inline tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/comment.fail.sh]: Please move the setup boilerplate to function that you can source and call? It's cleaner that way.

File: tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/comment.fail.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+dnf -y install sssd
```
Comment (dahaic) [inline tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/comment.fail.sh]: We don't want the test to work only on fedora, please change the logic to install package by appropriate packaging mechanism. You can get inspired by
`data/group_system/group_software/group_updating/group_updating_utils.sh`

File: linux_os/guide/services/sssd/sssd_offline_cred_expiration/oval/shared.xml (hunk 1 above)
Comment (dahaic) [reply linux_os/guide/services/sssd/sssd_offline_cred_expiration/oval/shared.xml]: I agree with Matus - explicit is better.

File: tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/wrong_value.fail.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# profiles = xccdf_org.ssgproject.content_profile_ospp
+
+SSSD_CONF="/etc/sssd/sssd.conf"
+dnf -y install sssd
+systemctl enable sssd
```
Comment (dahaic) [inline tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/wrong_value.fail.sh]: ditto

File: tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/correct_value.pass.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+# profiles = xccdf_org.ssgproject.content_profile_ospp
+
+SSSD_CONF="/etc/sssd/sssd.conf"
+
+dnf -y install sssd
+systemctl enable sssd
```
Comment (dahaic) [inline tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/correct_value.pass.sh]: ditto

File: tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/comment.fail.sh (hunk 3 above)
Comment (matusmarhefka) [reply tests/data/group_services/group_sssd/rule_sssd_offline_cred_expiration/comment.fail.sh]: I will use `yum` as it should work on both, Fedora and RHEL. IMO this should be solved by the test suite which should export some variable for installing and removing packages based on the tested VM.

**Comment by matusmarhefka** [comment]
Thanks for review @dahaic. It turned out that the original regex by @redhatrises used in OVAL check was not correct as it would match also options from subsequent sections after the `[pam]` section. I have corrected the regex and now all tests pass:
```
INFO - xccdf_org.ssgproject.content_rule_sssd_offline_cred_expiration
INFO - Script comment.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script wrong_value.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script wrong_section.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script correct_value.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
```

**Comment by dahaic** [comment]
Good stuff. Thank you @matusmarhefka
